com.unity.render-pipelines.core/Runtime/Lighting/ProbeVolume/ProbeBrickPool.cs

```diff
@@ -212,25 +258,51 @@ public static void FillDataLocation(ref DataLocation loc, SphericalHarmonicsL1[]
                             int iy = by + y;
                             int iz = bz + z;
 
-                            c.r = shl1[shidx].shAr[3];
-                            c.g = shl1[shidx].shAg[3];
-                            c.b = shl1[shidx].shAb[3];
-                            loc.TexL0.SetPixel(ix, iy, iz, c);
-
-                            c.r = shl1[shidx].shAr[0];
-                            c.g = shl1[shidx].shAr[1];
-                            c.b = shl1[shidx].shAr[2];
-                            loc.TexL1_R.SetPixel(ix, iy, iz, c);
-
-                            c.r = shl1[shidx].shAg[0];
-                            c.g = shl1[shidx].shAg[1];
-                            c.b = shl1[shidx].shAg[2];
-                            loc.TexL1_G.SetPixel(ix, iy, iz, c);
-
-                            c.r = shl1[shidx].shAb[0];
-                            c.g = shl1[shidx].shAb[1];
-                            c.b = shl1[shidx].shAb[2];
-                            loc.TexL1_B.SetPixel(ix, iy, iz, c);
+                            Vector3 L0 = SphericalHarmonicsL2Utils.GetCoefficient(shl2[shidx], 0);
+                            Vector3 L1R, L1G, L1B;
+                            SphericalHarmonicsL2Utils.GetL1(shl2[shidx], out L1R, out L1G, out L1B);
+
+                            // First texture will have L0 coefficients in RGB, and L1R.x in the alpha channel
+                            Color L0_L1Rx = new Color(L0.x, L0.y, L0.z, L1R.x);
+                            // Second texture will have L1_G coefficients in RGB and L1R.y in the alpha channel
+                            Color L1G_L1Ry = new Color(L1G.x, L1G.y, L1G.z, L1R.y);
+                            // Third texture will have L1_B coefficients in RGB and L1R.z in the alpha channel
+                            Color L1B_L1Rz = new Color(L1B.x, L1B.y, L1B.z, L1R.z);
+
+                            loc.TexL0_L1rx.SetPixel(ix, iy, iz, L0_L1Rx);
+                            loc.TexL1_G_ry.SetPixel(ix, iy, iz, L1G_L1Ry);
+                            loc.TexL1_B_rz.SetPixel(ix, iy, iz, L1B_L1Rz);
+
+                            if (bands == ProbeVolumeSHBands.SphericalHarmonicsL2)
```

Comment:
Not sure to understand why we go through all SH2 index
for (int brickIdx = 0; brickIdx < shl2.Length; brickIdx += kBrickProbeCountTotal)

and test 
if (bands == ProbeVolumeSHBands.SphericalHarmonicsL2)

(haven't read the full code, just seems weird)

Reply:
naively I will have except something like:

for (int brickIdx = 0; brickIdx < (bands == ProbeVolumeSHBands.SphericalHarmonicsL2) ? shl2.Length : shl1.Length; brickIdx += kBrickProbeCountTotal)

Reply:
The SH2 structure is always filled (to be able to not having to rebake in case a setting on asset is changed), but we load only the relevant bands to the runtime structures.